File: Algorithms/0938_Range_Sum_of_BST/Python/Range_Sum_of_BST_Solution_1.py

# Space: O(n)
# Time: O(n)

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root, low, high):
        if root.left is None and root.right is None: return root.val
        res = 0
        queue = [root]
        while queue:
            cur = queue.pop(0)
            if low<=cur.val<=high:
                res+=cur.val
            if cur.left: queue.append(cur.left)
            if cur.right: queue.append(cur.right)
        return res

File: Algorithms/0938_Range_Sum_of_BST/Java/RangeSumofBSTSolution1.java

// Space: O(n)
// Time: O(n)

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int rangeSumBST(TreeNode root, int low, int high) {
        if (root.left == null && root.right == null) return root.val;
        Queue<TreeNode> queue = new ArrayDeque();
        queue.add(root);
        int res = 0;
        while (!queue.isEmpty()) {
            TreeNode cur = queue.poll();
            if (low <= cur.val && cur.val <= high) res += cur.val;
            if (cur.left != null) queue.add(cur.left);
            if (cur.right != null) queue.add(cur.right);
        }
        return res;
    }
}





